examples/docker-image/main.tf

```
@@ -0,0 +1,72 @@
+module "label" {
+  source     = "cloudposse/label/null"
+  version    = "0.25.0"
+  attributes = [var.function_name]
+
+  context = module.this.context
+}
+
+module "ecr" {
+  source  = "cloudposse/ecr/aws"
+  version = "0.32.3"
+  name    = module.label.id
+
+  context = module.this.context
+}
+
+# Need to sleep for a few seconds after creating the ECR repository before we can push to it
+resource "time_sleep" "wait_15_seconds" {
+  count      = module.this.enabled ? 1 : 0
+  depends_on = [module.ecr]
+
+  create_duration = "15s"
+}
+
+# Cloud Posse does NOT recommend building and pushing images to ECR via Terraform code. This is a job for your CI/CD
+# pipeline. It is only done here for convenience and so that the example can be run locally.
+data "aws_region" "this" {}
+data "aws_caller_identity" "this" {}
```

Add these counts to keep the TF state empty when `enabled=false`

```suggestion
data "aws_region" "this" {
  count = local.enabled ? 1 : 0
}
data "aws_caller_identity" "this" {
  count = local.enabled ? 1 : 0
}
```